ui/src/customProtocolHandler.ts

```diff
@@ -0,0 +1,39 @@
+import * as ipc from "./ipc";
+import * as session from "./session";
+import * as modal from "./modal";
+import * as path from "./path";
+import * as error from "./error";
+import { inputStore } from "./search";
+
+export const register = (): void => {
+  ipc.listenCustomProtocolInvocation(message => {
+    session.waitUnsealed().then(() => {
+      handleMessage(message);
+    });
+  });
+};
+
+const handleMessage = (message: ipc.CustomProtocolInvocation) => {
+  const projectId = parseUrl(message.url);
+
+  if (projectId) {
+    inputStore.set(projectId);
+    modal.show(path.search());
+  } else {
+    error.show({
+      code: error.Code.CustomProtocolParseError,
+      message: "Could not extract project Radicle ID from the provided URL",
+      details: { url: message.url },
+    });
+  }
+};
+
+const parseUrl = (url: string): string | null => {
+  const match = url.match(/(rad:git:[\w]{59})/);
```

Comment:
We should probably be more strict about the parsing, document, this and write some unit tests.

Reply:
Agreed. I'm waiting for the [protocol discussion](https://github.com/radicle-dev/radicle-decisions/pull/10) to be settled before implementing the proper parser logic and tests.

Reply:
https://github.com/radicle-dev/radicle-upstream/pull/1652/commits/097bed52b267da37f9df942e94969af1d4219a66